api/model/storage/postgres/filter.go

```diff
@@ -439,16 +439,14 @@ func (s *Storage) buildCorrectnessResultWhere(wheres []string, params []interfac
 		for _, category := range f.Categories {
 			if strings.EqualFold(category, CorrectCategory) {
 				op = "="
-				break
 			} else if strings.EqualFold(category, IncorrectCategory) {
 				op = "!="
-				break
 			}
+			if op == "" {
```

Comment:
Since this check was moved inside the loop, `op := ""` should also be inside the loop.